### DISCUSSION TITLE
Djot to PDF conversion with SILE

### DISCUSSION THREAD

**Discussion by Omikhleia** [jgm/djot#236] [Show and tell]
I am glad announcing here a new release (v1.4.0) of the [markdown.sile](https://github.com/Omikhleia/markdown.sile) collection for the SILE typesetter, a set of packages aiming at making Markdown (initially) —and obviously **Djot** (now too)— first-class input languages in SILE, for direct rendering into a PDF.

---

Around one year ago, it initially started as an attempt overhauling Markdown support in SILE (hence the name of the collection) —using the ‡Lunamark library, with improvements proposed upstream, to obtain a direct AST to AST integration.

It was almost complete, as far as my typesetting needs go at least; but however good Lunamark is, it obviously lags beyond Pandoc in terms of features (esp. advanced tables), so on my way I also worked on a ‡Pandoc AST supporting package —for non technical readers, an alternate conversion route when one uses `pandoc -t json` and just feeds the output to SILE...

The latter too was nearing completion when, out of the wild, I heard about Djot... And as ‡Djot.lua came out, the incentive to include it and give it a serious try was strong. It was just the perfect moment to do so, and it got its way in v1.2.0, though at an experimental level.

However a specification-in-progress, Djot has some inherent qualities —a cleaner paradigm, less ambiguities and a great leaf-extension ability via symbols. To make that long story short, I'm now using it for most of my writing and typesetting workflow, with a great pleasure. This new version of the collection for SILE brings it close to my initial dreams of a simple but extensible input format.

---

‡ Lunamark, Pandoc, Djot.lua  —all of these were created by @jgm, so whatever I've achieved so far with all my little SILE experiments, I certainly have to acknowledge the debt here. Thanks for all of these!

**Comment by Omikhleia** [comment]
And just a little update, [_On some stars, flowers & places in Middle-earth_](https://sites.google.com/site/dragonbrumeux/on-some-stars-flowers-places-in-middle-earth), a small collection of Tolkien-related essays published earlier in November 2023, used the above solution and is (mostly)[^1] typeset in Djot (sources are [available](https://github.com/Omikhleia/awesome-sile-books/blob/main/dragon-de-brume-5/README.md) for those interested in digital typesetting and curious regarding how we did it).

[^1]: With a small bit of embedded Markdown for poetry verses, a few Djot extensions (epigraphs, templates, special symbols), a bit of external YAML for metadata and styles, and a very tiny bit of (SILE) Lua for fancy hacks.

**Reply by celtic-coder** [reply]
Excellent work! 👍

**Reply by simonmichael** [reply]
Thanks for the fine example and writing. I glanced at https://github.com/Omikhleia/awesome-sile-books/blob/main/dragon-de-brume-5/en/astre-rouge.dj to see djot in action and emerged from the beautiful PDF version of *On the red star near the Hunter’s Moon* only quite some time later !
